kafka-rest/src/main/java/io/confluent/kafkarest/resources/v2/PartitionsResource.java

```diff
@@ -15,53 +15,69 @@
-import org.slf4j.LoggerFactory;
-
-import java.util.List;
-import java.util.Vector;
-
-import javax.validation.Valid;
-import javax.validation.constraints.NotNull;
-import javax.ws.rs.Consumes;
-import javax.ws.rs.GET;
-import javax.ws.rs.POST;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.container.AsyncResponse;
-import javax.ws.rs.container.Suspended;
-import javax.ws.rs.core.Response;
+import static io.confluent.kafkarest.Utils.retrying;
 
-import io.confluent.kafkarest.KafkaRestContext;
 import io.confluent.kafkarest.Errors;
+import io.confluent.kafkarest.Errors.ConsumerInstanceNotFoundException;
+import io.confluent.kafkarest.KafkaRestConfig;
+import io.confluent.kafkarest.KafkaRestContext;
 import io.confluent.kafkarest.ProducerPool;
 import io.confluent.kafkarest.RecordMetadataOrException;
+import io.confluent.kafkarest.Utils;
 import io.confluent.kafkarest.Versions;
 import io.confluent.kafkarest.entities.AvroProduceRecord;
 import io.confluent.kafkarest.entities.BinaryProduceRecord;
+import io.confluent.kafkarest.entities.ConsumerInstanceConfig;
 import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.kafkarest.entities.GetTopicPartitionOffsetResponse;
 import io.confluent.kafkarest.entities.JsonProduceRecord;
 import io.confluent.kafkarest.entities.Partition;
 import io.confluent.kafkarest.entities.PartitionOffset;
 import io.confluent.kafkarest.entities.PartitionProduceRequest;
 import io.confluent.kafkarest.entities.ProduceRecord;
 import io.confluent.kafkarest.entities.ProduceResponse;
+import io.confluent.kafkarest.v2.KafkaConsumerState.OffsetTimestamp;
 import io.confluent.rest.annotations.PerformanceMetric;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.Vector;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Path("/topics/{topic}/partitions")
 @Produces({Versions.KAFKA_V2_JSON_BINARY_WEIGHTED_LOW, Versions.KAFKA_V2_JSON_AVRO_WEIGHTED_LOW,
            Versions.KAFKA_V2_JSON_WEIGHTED})
 @Consumes({Versions.KAFKA_V2_JSON})
-public class PartitionsResource {
-
+public final class PartitionsResource {
   private static final Logger log = LoggerFactory.getLogger(PartitionsResource.class);
 
+  private static final int CONSUMER_REQUEST_MAX_ATTEMPTS = 3;
+
   private final KafkaRestContext ctx;
+  private final String consumerGroup;
+
+  @GuardedBy("this")
+  private String consumerInstance = null;
```

Comment:
`consumerInstanceId` would be a better name i think

Reply:
also, there's no reason for `PartitionsResource` to explicitly know about/manage specific consumers either, this would be better abstracted away.